### ISSUE TITLE
Fix debug starchcat warnings

### ISSUE THREAD

**Issue by alexpreynolds** [bedops/bedops#134]
Compilation of debug build of ``starchcat`` raises ``clang++`` warnings:

```
starchcat.c:409:124: warning: use of old-style cast [-Wold-style-cast]
  ...-> %" PRId64" \t outFileSize -> %" PRIu64 " \t *cumulativeOutputSize -> %" PRIu64 "\n", (uint64_t) startOffset, (uint64_t) outFileS...
                                                                                             ^          ~~~~~~~~~~~
starchcat.c:409:148: warning: use of old-style cast [-Wold-style-cast]
  ...outFileSize -> %" PRIu64 " \t *cumulativeOutputSize -> %" PRIu64 "\n", (uint64_t) startOffset, (uint64_t) outFileSize, (uint64_t) *...
                                                                                                    ^          ~~~~~~~~~~~
starchcat.c:409:172: warning: use of old-style cast [-Wold-style-cast]
  ..." \t *cumulativeOutputSize -> %" PRIu64 "\n", (uint64_t) startOffset, (uint64_t) outFileSize, (uint64_t) *cumulativeOutputSize);
                                                                                                   ^          ~~~~~~~~~~~~~~~~~~~~~
starchcat.c:1846:57: warning: use of old-style cast [-Wold-style-cast]
        fprintf(stderr, "\tt_lineIdx -> %" PRIu64 "\n", (uint64_t) t_lineIdx);
                                                        ^          ~~~~~~~~~
starchcat.c:1847:69: warning: use of old-style cast [-Wold-style-cast]
        fprintf(stderr, "\tt_totalNonUniqueBases -> %" PRIu64 "\n", (uint64_t) t_totalNonUniqueBases);
                                                                    ^          ~~~~~~~~~~~~~~~~~~~~~
starchcat.c:1848:66: warning: use of old-style cast [-Wold-style-cast]
        fprintf(stderr, "\tt_totalUniqueBases -> %" PRIu64 "\n", (uint64_t) t_totalUniqueBases);
                                                                 ^          ~~~~~~~~~~~~~~~~~~
starchcat.c:3651:8: warning: use of old-style cast [-Wold-style-cast]
                            (uint64_t) mdRecord->mdOffset,
                            ^          ~~~~~~~~~~~~~~~~~~
starchcat.c:3655:8: warning: use of old-style cast [-Wold-style-cast]
                            (uint64_t) iter->size,
                            ^          ~~~~~~~~~~
starchcat.c:3656:8: warning: use of old-style cast [-Wold-style-cast]
                            (uint64_t) iter->lineCount);
                            ^          ~~~~~~~~~~~~~~~
9 warnings generated.
```

**Comment by alexpreynolds** [comment]
```
../../../../interfaces/src/data/starch/starchHelpers.c:2280:119: warning: use of old-style cast [-Wold-style-cast]
  ...fprintf(stderr, "\t(intermediate) duplicateElementExistsFlag: %d\tnestedElementExistsFlag: %d\n", (int) duplicateElementExistsFlag, (int) neste...
                                                                                                       ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:2280:153: warning: use of old-style cast [-Wold-style-cast]
  ...duplicateElementExistsFlag: %d\tnestedElementExistsFlag: %d\n", (int) duplicateElementExistsFlag, (int) nestedElementExistsFlag);
                                                                                                       ^     ~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:2370:120: warning: use of old-style cast [-Wold-style-cast]
  ..."\t(just-before-last-pass) duplicateElementExistsFlag: %d\tnestedElementExistsFlag: %d\n", (int) duplicateElementExistsFlag, (int) nestedElemen...
                                                                                                ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:2370:154: warning: use of old-style cast [-Wold-style-cast]
  ...duplicateElementExistsFlag: %d\tnestedElementExistsFlag: %d\n", (int) duplicateElementExistsFlag, (int) nestedElementExistsFlag);
                                                                                                       ^     ~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:2572:79: warning: use of old-style cast [-Wold-style-cast]
    fprintf(stderr, "\tWARNING:\nmdLength: %llu\nmd   - [%s]\nsha1 - [%s]\n", (unsigned long long) strlen(json), json, sha1Digest);
                                                                              ^                    ~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:3227:115: warning: use of old-style cast [-Wold-style-cast]
  ...fprintf(stderr, "\t(intermediate) duplicateElementExistsFlag: %d\tnestedElementExistsFlag: %d\n", (int) duplicateElementExistsFlag, (int) neste...
                                                                                                       ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:3227:149: warning: use of old-style cast [-Wold-style-cast]
  ...duplicateElementExistsFlag: %d\tnestedElementExistsFlag: %d\n", (int) duplicateElementExistsFlag, (int) nestedElementExistsFlag);
                                                                                                       ^     ~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:3346:111: warning: use of old-style cast [-Wold-style-cast]
  ...fprintf(stderr, "\t(intermediate) duplicateElementExistsFlag: %d\tnestedElementExistsFlag: %d\n", (int) duplicateElementExistsFlag, (int) neste...
                                                                                                       ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:3346:145: warning: use of old-style cast [-Wold-style-cast]
  ...duplicateElementExistsFlag: %d\tnestedElementExistsFlag: %d\n", (int) duplicateElementExistsFlag, (int) nestedElementExistsFlag);
                                                                                                       ^     ~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:3490:112: warning: use of old-style cast [-Wold-style-cast]
  ..."\t(last-pass) flag state: duplicateElementExistsFlag: %d\tnestedElementExistsFlag: %d\n", (int) duplicateElementExistsFlag, (int) nestedElemen...
                                                                                                ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:3490:146: warning: use of old-style cast [-Wold-style-cast]
  ...flag state: duplicateElementExistsFlag: %d\tnestedElementExistsFlag: %d\n", (int) duplicateElementExistsFlag, (int) nestedElementExistsFlag);
                                                                                                                   ^     ~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchHelpers.c:3558:79: warning: use of old-style cast [-Wold-style-cast]
    fprintf(stderr, "\tWARNING:\nmdLength: %llu\nmd   - [%s]\nsha1 - [%s]\n", (unsigned long long) strlen(json), json, sha1Digest);
                                                                              ^                    ~~~~~~~~~~~~
12 warnings generated.
```

**Comment by alexpreynolds** [comment]
```
End of search list.
../../../../interfaces/src/data/starch/starchMetadataHelpers.c:195:82: warning: use of old-style cast [-Wold-style-cast]
    fprintf(stderr, "\tduplicateElementExists: %d\n\tnestedElementExists: %d\n", (int) duplicateElementExists, (int) nestedElementExists);
                                                                                 ^     ~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchMetadataHelpers.c:195:112: warning: use of old-style cast [-Wold-style-cast]
    fprintf(stderr, "\tduplicateElementExists: %d\n\tnestedElementExists: %d\n", (int) duplicateElementExists, (int) nestedElementExists);
                                                                                                               ^     ~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchMetadataHelpers.c:587:69: warning: use of old-style cast [-Wold-style-cast]
            fprintf(stderr, "\titer->duplicateElementExists: %d\n", (int) iter->duplicateElementExists);
                                                                    ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchMetadataHelpers.c:588:66: warning: use of old-style cast [-Wold-style-cast]
            fprintf(stderr, "\titer->nestedElementExists: %d\n", (int) iter->nestedElementExists);
                                                                 ^     ~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../interfaces/src/data/starch/starchMetadataHelpers.c:981:61: warning: use of old-style cast [-Wold-style-cast]
        fprintf(stderr, "\tcurrentMdOffset: %" PRIu64 "\n", (uint64_t) currentMdOffset);
                                                            ^          ~~~~~~~~~~~~~~~
5 warnings generated.
```

**Comment by alexpreynolds** [comment]
Fixed in commit d8e4bdf
